DirectProgramming/DPC++FPGA/Tutorials/Features/ac_types/ac_int/src/ac_int.cpp

```diff
@@ -0,0 +1,350 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+// clang-format off
+#include <CL/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+// clang-format on
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities//include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+using namespace sycl;
+using namespace std;
+
+// Forward declare the kernel name in the global scope.
+// This is a FPGA best practice that reduces name mangling in the optimization
+// reports.
+class Add;
+class Div;
+class Mult;
+class ShiftLeft;
+class EfficientShiftLeft;
+class GetBitSlice;
+class SetBitSlice;
+
+using ac_int4 = ac_intN::int4;
+using ac_int14 = ac_intN::int14;
```

Comment:
I will let Paul weigh in on this one as well. I added these because they simplify the functions below and because I have seen non oneAPI code samples do it. I don't have a strong preference either way.

Reply:
Hi @paul-white-intel please review this part.

Reply:
I agree with Ajay. I would use them if you wanted to wrap a particular parameterization of `ac_int`, e.g. `ac_int<14, false>`, but if you're using the convenience type, then it's kind of redundant. What's the point of the convenience type then? :)

Reply:
To me, the fewer characters looked better. I didn't want to see ac_intN:: everywhere. If it doesn't add any value in your opinion, I will remove them.